sdk/core/Azure.Core/src/Serialization/AzureCoreExtensions.cs

```diff
@@ -65,27 +65,32 @@ public static class AzureCoreExtensions
         /// </summary>
         public static dynamic ToDynamicFromJson(this BinaryData utf8Json)
         {
-            return utf8Json.ToDynamicFromJson(DynamicDataOptions.Default);
+            DynamicDataOptions options = utf8Json is ResponseContent content ?
+                content.ProtocolOptions.GetDynamicOptions() :
+                DynamicDataOptions.Default;
+
+            return utf8Json.ToDynamicFromJson(options);
         }
 
         /// <summary>
         /// Return the content of the BinaryData as a dynamic type.
+        /// <paramref name="propertyNamingConvention">The naming convention to use for property names in the JSON content.</paramref>
         /// </summary>
-        public static dynamic ToDynamicFromJson(this BinaryData utf8Json, DynamicCaseMapping caseMapping, DynamicDateTimeHandling dateTimeHandling = DynamicDateTimeHandling.Rfc3339)
+        public static dynamic ToDynamicFromJson(this BinaryData utf8Json, PropertyNamingConvention propertyNamingConvention)
         {
-            DynamicDataOptions options = new()
-            {
-                CaseMapping = caseMapping,
-                DateTimeHandling = dateTimeHandling
-            };
+            DynamicDataOptions options = utf8Json is ResponseContent content ?
```

Comment:
Since this type is now internal, I'm wondering if we still actually need it, or if we can instead just hardcode the DateTime handling into DynamicData.

Reply:
Since it's internal, we can always remove it later.  If you feel strongly about it, I'm happy to add it to the uber tracking issue: https://github.com/Azure/azure-sdk-for-net/issues/33856